tests/rbd/readonly_namespace_image_migration.py

```diff
@@ -0,0 +1,162 @@
+"""Module to execute Live migration scenario on read-only namespace.
+
+Pre-requisites :
+We need cluster configured with atleast one client node with ceph-common package,
+conf and keyring files
+
+Test cases covered :
+CEPH-83573341 - Set read only on namespace and migrate the images and
+verify migration is not allowed
+
+Test Case Flow :
+1. Create pool.
+2. Create a namespace within the pool
+3. Give readonly access to the namespace
+4. Create images within the namespace
+5. Migrate the created images to other pool
+6. verify that image migration is not allowed as image is read-only.
+"""
+
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import Rbd, initial_rbd_config, verify_namespace_exist
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def image_migrate_and_verify(
+    rbd,
+    src_pool,
+    dest_pool,
+    namespace,
+    image,
+):
+    """Migrate the image from source to destination and verify migration
+
+    Args:
+        src_pool: source pool name
+        dest_pool: destination pool name
+        namespace: name of namespace
+    """
+    src_spec1 = src_pool + "/" + namespace + "/" + image
+    dest_spec1 = dest_pool + "/" + image
+    client_id = "client.1"
+    cmd1 = "ceph auth caps client.1"
+    cmd1 += " mon 'allow *' mds 'allow *'"
+    cmd1 += f" mgr 'profile rbd-read-only pool={src_pool}"
+    cmd1 += f" namespace={namespace}' osd 'profile rbd-read-only'"
+    rbd.exec_cmd(cmd=cmd1)
+    cmd2 = "ceph auth get client.1 -o /etc/ceph/ceph.client.1.keyring"
+    rbd.exec_cmd(cmd=cmd2)
+    rbd.create_image(
+        pool_name=src_pool,
+        image_name=image,
+        size="5G",
+        namespace=namespace,
+    )
+    rbd.image_exists(
+        pool_name=src_pool,
+        image_name=image,
+        namespace=namespace,
+    )
+    out, err = rbd.migration_prepare(src_spec1, dest_spec1, client_id=client_id)
+    if "Migration: prepare: failed to open image" in err:
+        log.info(
+            "Image migration for read-only images from namespace is successfully blocked"
+        )
+        return 0
+    else:
+        log.error(f"{err}")
+        log.error(
+            "Image migration is allowed even though image is read-only in namespace"
```

Comment:
[optional]
Can we convey -
"Image is read only for client X" and "migration blocked from client X"

Reply:
done